tests/programs/_build_clang_native.sh

```diff
@@ -0,0 +1,27 @@
+set -ex
+
+ROOT_DIR=$(pwd)
+CC="clang-18"
```

Comment:
In the spirit that someone might copy this script, I recommend using tweakable variables, see `CLANG` and `CLANGXX` in this [file](https://github.com/xxuejie/ckb-libcxx-builder/blob/main/build.sh#L4) for an example.